From 5f9962f5d0e50c63e150ddc3d6d34beb1c755e83 Mon Sep 17 00:00:00 2001
From: LuK1337 <priv.luk@gmail.com>
Date: Mon, 11 Jan 2016 17:40:58 +0100
Subject: [PATCH 063/206] Move mApplicationContext to init()

Fixes following NPE:
01-11 17:36:42.607  4329  4391 E AndroidRuntime: FATAL EXCEPTION: Thread-235
01-11 17:36:42.607  4329  4391 E AndroidRuntime: Process: org.cyanogenmod.snap, PID: 4329
01-11 17:36:42.607  4329  4391 E AndroidRuntime: java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.res.Resources android.content.Context.getResources()' on a null object reference
01-11 17:36:42.607  4329  4391 E AndroidRuntime: 	at com.android.camera.PhotoModule.updateCameraParametersPreference(PhotoModule.java:3492)
01-11 17:36:42.607  4329  4391 E AndroidRuntime: 	at com.android.camera.PhotoModule.setCameraParameters(PhotoModule.java:3714)
01-11 17:36:42.607  4329  4391 E AndroidRuntime: 	at com.android.camera.PhotoModule.startPreview(PhotoModule.java:2715)
01-11 17:36:42.607  4329  4391 E AndroidRuntime: 	at com.android.camera.PhotoModule.-wrap14(PhotoModule.java)
01-11 17:36:42.607  4329  4391 E AndroidRuntime: 	at com.android.camera.PhotoModule$OpenCameraThread.run(PhotoModule.java:265)

Change-Id: I6ca0dfe402cab46dd7b6f102921062f3c7ec8d43
---
 src/com/android/camera/PhotoModule.java | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/com/android/camera/PhotoModule.java b/src/com/android/camera/PhotoModule.java
index c5e8ce8e5..c02102045 100644
--- a/src/com/android/camera/PhotoModule.java
+++ b/src/com/android/camera/PhotoModule.java
@@ -593,6 +593,7 @@ public class PhotoModule
         mCameraId = getPreferredCameraId(mPreferences);
 
         mContentResolver = mActivity.getContentResolver();
+        mApplicationContext = CameraApp.getContext();
 
         // Surface texture is from camera screen nail and startPreview needs it.
         // This must be done before startPreview.
@@ -934,8 +935,6 @@ public class PhotoModule
             return;
         }
 
-        mApplicationContext = CameraApp.getContext();
-
         // Initialize location service.
         boolean recordLocation = RecordLocationPreference.get(mPreferences,
                 CameraSettings.KEY_RECORD_LOCATION);
-- 
2.17.1

